homeassistant/components/switch/fritzdect.py

```diff
@@ -0,0 +1,151 @@
+"""
+Support for FRITZ!DECT Switches.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/switch.fritzdect/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.switch import (SwitchDevice, PLATFORM_SCHEMA)
+from homeassistant.const import (
+    CONF_HOST, CONF_PASSWORD, CONF_USERNAME)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import TEMP_CELSIUS, STATE_UNKNOWN
+
+REQUIREMENTS = ['fritzhome==1.0.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+# Standard Fritz Box IP
+DEFAULT_HOST = 'fritz.box'
+
+ATTR_CURRENT_CONSUMPTION = 'Current Consumption'
+ATTR_CURRENT_CONSUMPTION_UNIT = 'W'
+
+ATTR_TOTAL_CONSUMPTION = 'Total Consumption'
+ATTR_TOTAL_CONSUMPTION_UNIT = 'kWh'
+
+ATTR_TEMPERATURE = 'Temperature'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST, default=DEFAULT_HOST): cv.string,
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Add all switches connected to Fritz Box."""
+    from fritzhome.fritz import FritzBox
+
+    host = config.get(CONF_HOST)
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+
+    # Log into Fritz Box
+    fritz = FritzBox(host, username, password)
+    fritz.login()
+
+    # Add all actors to hass
+    for actor in fritz.get_actors():
+        # Only add devices that support switching
+        if actor.has_switch:
+            data = FritzDectSwitchData(fritz, actor.actor_id)
+            add_devices([FritzDectSwitch(hass, data, actor.name)], True)
+
+
+class FritzDectSwitch(SwitchDevice):
+    """Representation of a FRITZ!DECT switch."""
+
+    def __init__(self, hass, data, name):
+        """Initialize the switch."""
+        self.units = hass.config.units
+        self.data = data
+        self._name = name
+
+    @property
+    def name(self):
+        """Return the name of the FRITZ!DECT switch, if any."""
+        return self._name
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the device."""
+        attrs = {}
+
+        if self.data.has_powermeter:
+            attrs[ATTR_TOTAL_CONSUMPTION] = "%.3f %s" % \
+              (float(self.data.total_consumption), ATTR_TOTAL_CONSUMPTION_UNIT)
+
+        if self.data.has_temperature:
+            attrs[ATTR_TEMPERATURE] = "%.1f %s" % \
+              (self.units.temperature(self.data.temperature, TEMP_CELSIUS),
+               self.units.temperature_unit)
```

Comment:
It looks like that adding the unit of measurement will make it hard to use the attributes in templates and for automations.